### ISSUE TITLE
Using both letsencrypt & buypass ?

### ISSUE THREAD

**Issue by centminmod** [Neilpang/acme.sh#2007]
From buypass ca instructions at https://github.com/Neilpang/acme.sh/wiki/BuyPass.com-CA, you need to register the email account first. Where does this get saved exactly ? In `/root/.acme.sh/account.conf ` ? I ask as I am curious how this would work if I am trying get both letsencrypt and buypass ssl certs working on the same server.

So say I issued a letsencrypt ssl cert first but didn't set up a registered email with letsencrypt

```
/root/.acme.sh/acme.sh --force --issue --days 60 -d domain.com -d www.domain.com -w /home/nginx/domains/domain.com/public -k 2048 --useragent centminmod-centos-acmesh-webroot --log /root/centminlogs/acmetool.sh-debug-log-domain.com.log --log-level 2
```
```
/root/.acme.sh/acme.sh --installcert -d domain.com -d www.domain.com --certpath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.cer --keypath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.key --capath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.cer --reloadCmd /usr/bin/ngxreload --fullchainpath /usr/local/nginx/conf/ssl/domain.com/domain.com-fullchain-acme.key
```

but then I wanted to use buypass ssl cert with same `domain.com` 

```
/root/.acme.sh/acme.sh --server https://api.buypass.com/acme/directory --register-account  --accountemail me@example.com
```
```
/root/.acme.sh/acme.sh --server https://api.buypass.com/acme/directory --force --issue --days 165 -d domain.com -d www.domain.com -w /home/nginx/domains/domain.com/public -k 2048 --useragent centminmod-centos-acmesh-webroot --log /root/centminlogs/acmetool.sh-debug-log-domain.com.log --log-level 2
```
```
/root/.acme.sh/acme.sh --installcert -d domain.com -d www.domain.com --certpath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.cer --keypath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.key --capath /usr/local/nginx/conf/ssl/domain.com/domain.com-acme.cer --reloadCmd /usr/bin/ngxreload --fullchainpath /usr/local/nginx/conf/ssl/domain.com/domain.com-fullchain-acme.key
```

1. Will the buypass registered email address be saved in a variable which letsencrypt would also pick up too even if I didn't set/register an email for letsencrypt ?
2. Does buypass do anything for `--useragent ` flag like with letsencrypt ?

**Comment by Neilpang** [comment]
>> 1. Will the buypass registered email address be saved in a variable 

Yes for now,  but I'm thinking about that:  whether we should use different email for different CA.  what do you think ?


>> 2. Does buypass do anything for --useragent flag like with letsencrypt ?

Yes,  the useragent is shared by all the servers.

>> 3.  Does buypass have a staging test ENDPOINT url too ?

Sorry,  I don't find that yet.

**Comment by centminmod** [comment]
cheers

> Yes for now, but I'm thinking about that: whether we should use different email for different CA. what do you think ?

Yeah having a different variable for buypass email from letsencrypt would be nice as usually i don't set letsencrypt email so using buypass for email rego only would be preferred option.

So as i understand it if letsencrypt and buypass both use registered emails, and you're switching between letsencrypt and buypass ssl cert issuance, you would get 2 sets of emails when ssl cert renewal is up - one from letsencrypt and one from buypass for the same domain ?

**Comment by Neilpang** [comment]
currently. we save the account email in  `~/.acme.sh/account.conf`,  which means it's shared by all CA.
Maybe we should save the email in  `~/.acme.sh/ca/<CA_NAME>/ca.conf` file,  which is only valid for the specified CA.

It uses the same variable `--accountemail`, but only save in the CA specific conf.



>> you would get 2 sets of emails when ssl cert renewal is up

Yes, I guess so.

**Comment by centminmod** [comment]
> Maybe we should save the email in `~/.acme.sh/ca/<CA_NAME>/ca.conf` file, which is only valid for the specified CA.

I believe that would be the best way - would scale well when even more acme compliant CAs come out in the future :)

Or something like aws-cli's named profiles feature might be another method i.e. https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-profiles.html

**Comment by baoang** [comment]
So, @Neilpang , when will you release the separate-account-featured version, by putting accounts into their corresponding dirs, or perhaps keeping them in the existing account.conf with relevant variables/prefixes?

```
SAVED_CF_Key='12345上山打老虎'
SAVED_CF_Email='67890@老虎不在.家'
SAVED_NAMECHEAP_KEY='54321天王盖地虎'
SAVED_NAMECHEAP_EMAIL='98760@宝塔镇河妖'
BUYPASS_ACCOUNT_EMAIL='ilovebp@buypass.com'
LE_ACCOUNT_EMAIL='ilovelemore@letsencrypt.org'
```

**Comment by neilpang** [comment]
Closing as stale: no activity for over 18 months. If this still applies on the current release, please open a new issue with the `--debug 2` log.

**Comment by neilpang** [comment]
Implemented in 389518e1: the account email is now stored per-CA in ca/<host>/ca.conf (CA_EMAIL), no longer shared across CAs.
